curation-migration-datasync/database/dataService.ts

```diff
@@ -130,6 +132,78 @@ export class DataService {
     });
   }
 
+  /**
+   * updates the curated_feed_prospects table and curated_feed_queued_items
+   * fields, if they are set in the eventBody.
+   * won't update the fields if they are set to null in the eventBody.
+   * @param eventBody event body
+   * @param curatedRecId curatedRecId corresponding to the approvedItem's externalId
+   * @param domainId domain id from parser, optional field set only when
+   *        publisher is not null in the event body.
+   */
+  public async updateApprovedItem(
+    eventBody: ApprovedItemPayload,
+    curatedRecId: number
+  ): Promise<void> {
+    const item = await this.db(config.tables.curatedFeedItems)
+      .select()
+      .join(
+        config.tables.curatedFeedProspects,
+        'curated_feed_prospects.prospect_id',
+        'curated_feed_items.prospect_id'
+      )
+      .join(
+        config.tables.curatedFeedQueuedItems,
+        'curated_feed_queued_items.queued_id',
+        'curated_feed_items.queued_id'
+      )
+      .where('curated_rec_id', curatedRecId)
+      .first();
+
+    let topicId;
+    if (eventBody.topic) {
+      topicId = await this.getTopicIdByName(
+        getTopicForReaditLaTmpDatabase(eventBody.topic)
+      );
+    } else {
+      topicId = item['topic_id'];
+    }
+
+    const curator = eventBody.createdBy
+      ? getCuratorNameFromSso(eventBody.createdBy)
+      : null;
+
+    //update curated_feed_prosects and curated_feed_queued_items fields
+    //if corresponding eventBody field is set,
+    //otherwise set them to what's existing in the database
+    await this.db.transaction(async (trx) => {
+      await trx(config.tables.curatedFeedProspects)
+        .update({
```

Comment:
nit: We don't have to get the previous record -- we can do a partial update only of fields that are set